src/adapters/quantcast.js

```diff
@@ -0,0 +1,122 @@
+const utils = require('../utils.js');
+const bidfactory = require('../bidfactory.js');
+const bidmanager = require('../bidmanager.js');
+const ajax = require('../ajax.js');
+const QUANTCAST_CALLBACK_URL = 'http://global.qc.rtb.quantserve.com:8080/qchb';
+
+var QuantcastAdapter = function QuantcastAdapter() {
+
+  const BIDDER_CODE = 'quantcast';
+
+  const DEFAULT_BID_FLOOR = 0.0000000001;
+  // The following 2 constants are adopted from bidfactory.js codes
+  const BID_STATUS_CODE_AVAILABLE = 1;
+  const BID_STATUS_CODE_EMPTY = 2;
+  let bidRequests = {};
+
+  let returnEmptyBid = function() {
+      var bidsRequested = $$PREBID_GLOBAL$$._bidsRequested.find(bidSet => bidSet.bidderCode === BIDDER_CODE).bids;
+      if (bidsRequested.length > 0) {
+        let bid = bidfactory.createBid(BID_STATUS_CODE_EMPTY);
+        bid.bidderCode = BIDDER_CODE;
+        bidmanager.addBidResponse(bidsRequested[0].placementCode, bid);
+      }
+      return;
+    };
+
+  //expose the callback to the global object:
+  $$PREBID_GLOBAL$$.handleQuantcastCB = function (responseText) {
+    if(utils.isEmpty(responseText)) {
+      return;
+    }
+    let response = JSON.parse(responseText);
+    if(typeof(response) === 'undefined' || !response.hasOwnProperty('bids') || utils.isEmpty(response.bids)) {
+      return returnEmptyBid();
+    }
+
+    for(let i = 0; i < response.bids.length; i++) {
+      let seatbid = response.bids[i];
+      let key = seatbid.placementCode + "-" + seatbid.width + 'x' + seatbid.height;
+      var request = bidRequests[key];
+      if(request === null) {
+        return returnEmptyBid();
+      }
+      // This line is required since this is the field
+      // that bidfactory.createBid looks for
+      request.bidId = request.imp[0].placementCode;
+      let responseBid = bidfactory.createBid(BID_STATUS_CODE_AVAILABLE, request);
+
+      responseBid.cpm = seatbid.cpm;
+      responseBid.ad = seatbid.ad;
+      responseBid.height = seatbid.height;
+      responseBid.width = seatbid.width;
+      responseBid.bidderCode = response.bidderCode;
+
+      bidmanager.addBidResponse(request.bidId, responseBid);
+    }
+
+  };
+
+  function callBids(params) {
+    let bids = params.bids || [];
+    if (bids.length === 0) {
+      return;
+    }
+
+    let referrer = utils.getTopWindowUrl();
+    let loc = utils.getTopWindowLocation();
+    let domain = loc.hostname;
+    let publisherId = 0;
+
+    publisherId = '' + bids[0].params.publisherId;
+    utils._each(bids, function(bid) {
+      // make sure the "sizes" are an array of arrays
+      if (!(bid.sizes[0] instanceof Array)) {
```

Comment:
DFP allows sizes to be array and array of arrays. Support both.
parseSizesInput function https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L117 will help